### ISSUE TITLE
Wrong decoding of memory operand sizes

### ISSUE THREAD

**Issue by BenBE** [vmt/udis86#51]
Given the example x64 instruction "83 3d 29 d4 20 00 00" I get the following result in Intel syntax:

cmp dword ptr [rip+0x20d429], 0x00

While it IS correct that the argument in the displacement is 32 bit inside the instruction this decoding of the instruction doesn't reflect the fact that the memory access is on byte level and thus

cmp byte ptr [rip+0x20d429], 0x00

is the correct decoding of this instruction. The word ptr and dword ptr versions would be using primary opcode 0x81. with and without operand size prefix.

TL;DR: The operand size designation byte/word/dword should reflect the size of accessed memory, not the size of the used address offset.

**Comment by vmt** [comment]
The "dword" in,

```
$ echo "83 3d 29 d4 20 00 00" | udcli/udcli -64 -x
0000000000000000 833d29d4200000   cmp dword [rip+0x20d429], 0x0
```

is the size of the first operand used in the comparison, not the size of the address offset.
